storage/myisam/myisamchk.c

```diff
@@ -1397,7 +1397,7 @@ static void descript(HA_CHECK *param, register MI_INFO *info, char * name)
 	 key < share->state.header.uniques; key++, uniqueinfo++)
     {
       my_bool new_row=0;
-      char null_bit[8],null_pos[8];
+      char null_bit[8],null_pos[10];
```

Comment:
I believe there's no real problem here, since null_pos is normally at the beginning of a record and shouldn't be larger than 512 or so. But code wise I agree with gcc 7.

Per my understanding you need at least 11 bytes to store 32bit signed integer in decimal form plus 1 byte for trailing 0. Where does 10 come from and why it works?

I would suggest to rewrite this code so that these variables are optimised away. That is split printf() in 3 parts and print directly into stdio buffer. It should improve code readability and be better both performance wise.